Common/src/main/java/com/tabnineCommon/lifecycle/PluginInstalled.kt

```diff
@@ -0,0 +1,41 @@
+package com.tabnineCommon.lifecycle
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.util.messages.MessageBusConnection
+import com.intellij.util.messages.Topic
+
+fun interface PluginInstalled {
+    companion object {
+        val PLUGIN_INSTALLED_TOPIC = Topic.create(
+            "Plugin Installed Notifier",
+            PluginInstalled::class.java
+        )
+        var isNewInstallation: Boolean? = null // denotes unknown
+            set(value) {
+                // once toggled to true , don't allow  toggling back to false
+                // if null allow toggling to false
+                // don't double publish
+                if (field != true && field != value) {
+                    field = value
+                    publish()
+                }
+            }
+
+        private fun publish() {
```

Comment:
you can have `publish` accept the new value, and this way you don't have to use `isNewInstallation?.let`